src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -592,7 +628,7 @@ protected void closeInternal() {
          * @param oldCurrent a new read only reader for the old current (should replace the previous reference)
          * @param newCurrent a reader into the new current.
          */
-        synchronized void onNewTranslog(ChannelReader oldCurrent, ChannelReader newCurrent) throws IOException {
+        synchronized void onNewTranslog(TranslogReader oldCurrent, TranslogReader newCurrent) throws IOException {
             // even though the close method removes this view from outstandingViews, there is no synchronisation in place
             // between that operation and an ongoing addition of a new translog, already having an iterator.
             // As such, this method can be called despite of the fact that we are closed. We need to check and ignore.
```

Comment:
I think we can merge View and FSView (not here, later!)... seems like an abstraction layer that we don't need.